asistente_ladm_col/utils/qgis_utils.py

```diff
@@ -1041,24 +1046,118 @@ def show_etl_model(self, db, input_layer, ladm_col_layer_name, geometry_type=Non
+                                                "txt"))
+
+        field_mapping_list = []
+
+        with open(path_file_field_mapping) as file_field_mapping:
+            try:
+                mapping = ast.literal_eval(file_field_mapping.read())
+            except:
+                mapping = None
+
+        return mapping
+
+    def func_save_field_mapping(self,ladm_col_layer_name):
+        files_mapping_path = os.path.join(os.path.expanduser('~'), 'Asistente-LADM_COL', 'field_mapping')
+        if not os.path.exists(files_mapping_path):
+            os.makedirs(files_mapping_path)
+
+        log_path = os.path.join(processing.tools.system.userFolder(),
+                    'processing.log')
+
+        if not os.path.exists(FIELD_MAPPING_PATH):
+            os.makedirs(FIELD_MAPPING_PATH)
+
+        log_path =  os.path.join(processing.tools.system.userFolder(),
+                    'processing.log')
+
+        with open(log_path) as log_file:
+            log_file = log_file.read().split("ALGORITHM")
+            log_file = log_file[len(log_file)-1]
+            log_file = log_file.split("mapping")
+            log_file = log_file[len(log_file)-1]
+            log_file = log_file.split("output")
+            log_file = log_file[0]
+            log_file = log_file.strip("':")
+            log_file = log_file.strip(",'")
+
+        name_field_mapping = "{}_{}.{}".format(ladm_col_layer_name,
+        datetime.datetime.now().strftime("%Y%m%d_%H_%M_%S"),
+        "txt")
+        txt_field_mapping_path = os.path.join(FIELD_MAPPING_PATH,
+						                          name_field_mapping)
+
+        QgsApplication.messageLog().logMessage("Save recent mapping: {}".format(name_field_mapping), PLUGIN_NAME, Qgis.Info)
+
+        with open(txt_field_mapping_path,"w+") as file:
+        	file.write(log_file)
+        	file.close()
+
+    def fields_mapping(self, layer_name):
+
+        combox_data = []
+
+        files = glob.glob(os.path.join(FIELD_MAPPING_PATH, "{}_{}{}".format(layer_name, '[0-9]'*8, "*")))
+        files.sort(key=lambda path: os.path.getmtime(path))
+
+        if (len(files) > 10):
+            for path in files[0:len(files)-10]:
+                os.remove(path)
+                files = glob.glob(os.path.join(FIELD_MAPPING_PATH, "{}_{}{}".format(layer_name, '[0-9]'*8, "*")))
+                files.sort(key=lambda path: os.path.getmtime(path))
+
+        files.reverse()
+        for file in files:
+            combox_data.append(os.path.basename(file).strip(".txt"))
+
+        return combox_data
+
+    def replace_field_mapping(self, name, layer_name):
+        files = glob.glob(os.path.join(FIELD_MAPPING_PATH, "{}_{}{}".format(layer_name, '[0-9]'*8, "*")))
+        files.sort(key=lambda path: os.path.getmtime(path))
+        name = "{}.{}".format(name, "txt")
+
+        for path in files:
+            if path.endswith(name):
+                os.remove(path)
```

Comment:
If I understand well, we don't need to iterate through files. We know the base path for all field mappings and the name of the field mapping we want to delete. Just get the full path of this specific field mapping and delete it.